backend/capellacollab/sessions/operators/k8s.py

```diff
@@ -885,6 +857,92 @@ def _get_pods(self, label_selector: str) -> list[client.V1Pod]:
             namespace=namespace, label_selector=label_selector
         ).items
 
+    def list_files(self, _id: str, directory: str, show_hidden: bool):
+        def print_file_tree_as_json():
+            import json  # pylint: disable=redefined-outer-name,reimported
+            import pathlib
+            import sys
+
+            print("Using CLI arguments: " + str(sys.argv[1:]), file=sys.stderr)
+
+            def get_files(dir: pathlib.PosixPath, show_hidden: bool):
+                file = {
+                    "path": str(dir.absolute()),
+                    "name": dir.name,
+                    "type": "directory",
+                    "children": [],
+                }
+
+                assert isinstance(file["children"], list)
+
+                for item in dir.iterdir():
+                    if not show_hidden and item.name.startswith("."):
+                        continue
+                    if item.is_dir():
+                        file["children"].append(get_files(item, show_hidden))
+                    elif item.is_file():
+                        file["children"].append(
+                            {
+                                "name": item.name,
+                                "path": str(item.absolute()),
+                                "type": "file",
+                            }
+                        )
+
+                return file
+
+            print(
+                json.dumps(
+                    get_files(
+                        pathlib.Path(sys.argv[1]), json.loads(sys.argv[2])
+                    )
+                )
+            )
+
+        source = helper.get_source_of_python_function(print_file_tree_as_json)
+
+        pod_name = self._get_pod_name(_id)
+        # We have to use subprocess to get it running until this issue is solved:
+        # https://github.com/kubernetes/kubernetes/issues/89899
+        # Python doesn't start evaluating the code before EOF, but there is no way to close stdin
+
+        try:
+            response = subprocess.run(
+                ["kubectl"]
+                + self.kubectl_arguments
+                + [
+                    "--namespace",
+                    namespace,
+                    "exec",
+                    "--stdin",
+                    pod_name,
+                    "--container",
+                    _id,
+                    "--",
+                    "python",
+                    "-",
+                    directory,
+                    json.dumps(show_hidden),
+                ],
```

Comment:
Hmm.. At this point we could just as well use `ls` :smile: (but let's not change that now)

Reply:
Parsing ls is a lot more pain then just running Python. We went there, whitespaces, symlinks, remote code execution (when running `ls` on a directory called `$(date)` for example.